### PR TITLE
SPR-157 Delete Confirmation Prompt

### PR DESCRIPTION
**Pull request by dbarkowsky**
# Description

This PR includes the following proposed change(s):

- Added a deletion/removal prompt when user tries to delete a record, remove an approval, or remove a file.
- Fixed a bug where removing a file would cause files to cascade up.

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change
- [ ] Dependencies added/removed

## Aspect(s) of Project Affected
- [x] Frontend
- [ ] API
- [ ] Database
- [ ] Workflows
- [ ] Documentation

## Best Practices Checklist
- [x] I have performed a self-review of my own code
- [x] My code follows the Airbnb React Style Guidelines
- [x] Documentation has been updated to reflect my changes
- [x] New and existing tests pass locally with my changes

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
Tested manually on local build.

### CODE REVIEW

File: app/src/components/custom/modals/DeletePrompt.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  blurb: string;
+  id: string;
+}
+
+/**
+ * @description A modal element that can be used to confirm deletions.
+ * @param {DeletePromptProps} props The properties passed to the component.
+ * @returns A React component.
+ */
+const DeletePrompt = (props: DeletePromptProps) => {
```
[the remark this answers is not in the document]
Comment (dbarkowsky) [reply app/src/components/custom/modals/DeletePrompt.tsx:25]: This is too low of a threshold, but I can't see where to change it. Turning it off for now.

File: app/src/components/custom/modals/DeletePrompt.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      <div
+        style={{
+          display: 'flex',
+          justifyContent: 'space-between',
+          marginTop: '2em',
+        }}
+      >
+        <ActionButton
+          style={buttonStyles.secondary}
+          handler={() => {
```
[the remark this answers is not in the document]
Comment (dbarkowsky) [reply app/src/components/custom/modals/DeletePrompt.tsx:61]: Not similar enough to change. Considering this a false positive.

File: app/src/components/custom/forms/RequestForm.tsx [hunk 3]
```diff
@@ -134,6 +140,12 @@ const RequestForm = (props: RequestFormProps) => {
 
   return (
     <>
+      <DeletePrompt
+        deleteHandler={handleDelete}
+        title='Delete Request?'
+        blurb='Are you sure you want to delete this request?;;This action will hide the request from the requestor and may only be undone by an administrator.'
```
Comment (BradyMitch) [inline app/src/components/custom/forms/RequestForm.tsx:146]: You could also use a string array for blurb

File: app/src/components/custom/modals/DeletePrompt.tsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+/**
+ * @description A modal element that can be used to confirm deletions.
+ * @param {DeletePromptProps} props The properties passed to the component.
+ * @returns A React component.
+ */
+const DeletePrompt = (props: DeletePromptProps) => {
+  const { deleteHandler, title, blurb, id } = props;
+  return (
+    <dialog
```
Comment (BradyMitch) [inline app/src/components/custom/modals/DeletePrompt.tsx:28]: I like the use of the dialog element as it's had some big improvements recently.

**Review by BradyMitch — APPROVED** [review]

File: app/src/components/custom/forms/RequestForm.tsx (hunk 3 above)
Comment (dbarkowsky) [reply app/src/components/custom/forms/RequestForm.tsx:146]: That would have been better. I will add this to the to-do list.

**Pull request merged**